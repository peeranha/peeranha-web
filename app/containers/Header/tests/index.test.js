import { Header, mapDispatchToProps } from '../index';

const cmp = new Header();
cmp.props = {};
cmp.props = {
  account: 'user',
  userIsInSystem: false,
  showSignUpModalDispatch: jest.fn(),
  showLoginModalDispatch: jest.fn(),
};

describe('<Header />', () => {
  it('mapDispatchToProps test', () => {
    const test = 'test';
    const dispatch = () => test;

    expect(typeof mapDispatchToProps(dispatch) === 'object').toBe(true);
    expect(mapDispatchToProps(dispatch).dispatch).toBe(dispatch);
    expect(mapDispatchToProps(dispatch).showSignUpModalDispatch()).toBe(test);
    expect(mapDispatchToProps(dispatch).showLoginModalDispatch()).toBe(test);
  });

  it('render', () => {
    expect(cmp.render()).toMatchSnapshot();
  });
});
