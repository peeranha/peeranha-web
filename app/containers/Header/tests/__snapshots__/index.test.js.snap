// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshot test 1`] = `
<InjectIntl(Component)
  account="account"
  expandLeftMenuNavigation={[MockFunction]}
  isMenuVisible={false}
  profileInfo={Object {}}
  showLoginModalDispatch={[MockFunction]}
  showMenu={[MockFunction]}
/>
`;
